### PR TITLE
Tighten practitioner boundaries and enforce service windows

### PR DESCRIPTION
**Pull request by wasafiri**
- Enforce service-window eligibility for training requests and practitioner assignments
- Add Application#service_window_active? using the existing waiting-period Policy
- Prevent expired applications from requesting training or being assigned trainers/evaluators
- Remove admin-owned training scheduling and completion actions from the application workflow (trainers schedule and complete training sessions, not admins)
- Restrict session mutations to the assigned trainer/evaluator while preserving read-only access for admins
- Update admin and practitioner views
- Add regression tests for service-window checks, route removal, and practitioner authorization boundaries

### CODE REVIEW

File: app/models/concerns/evaluation_management.rb [hunk 1]
```diff
@@ -5,10 +5,18 @@
 module EvaluationManagement
   extend ActiveSupport::Concern
 
+  EVALUATION_SERVICE_WINDOW_ERROR = 'This application is outside the service window for evaluation services.'
```
Comment (mepps) [inline app/models/concerns/evaluation_management.rb]: I don't think it's a good idea for translatability for an english string to be hard-coded into the model.

File: app/models/concerns/training_management.rb [hunk 2]
```diff
@@ -5,10 +5,18 @@
 module TrainingManagement
   extend ActiveSupport::Concern
 
+  TRAINING_SERVICE_WINDOW_ERROR = 'This application is outside the service window for training services.'
```
Comment (mepps) [inline app/models/concerns/training_management.rb]: Same as above

File: app/models/application.rb [hunk 3]
```diff
@@ -516,6 +537,24 @@ def voucher_issuable?
...
+      !active_training_session_present? &&
+      training_sessions.where(created_at: training_requested_at..).none?
+  end
+
+  def service_window_active?
+    return false unless status_approved?
+    return false if application_date.blank?
+
+    waiting_period = Policy.get('waiting_period_years') || 3
+    application_date.to_date + waiting_period.years > Date.current
```
Comment (mepps) [inline app/models/application.rb:555]: application_date is the date of the original application or when it was approved?

Comment (wasafiri) [reply app/models/application.rb:555]: of the original application

File: app/models/concerns/evaluation_management.rb (hunk 1 above)
Comment (wasafiri) [reply app/models/concerns/evaluation_management.rb]: Agreed, will fix

File: app/models/concerns/training_management.rb (hunk 2 above)
Comment (wasafiri) [reply app/models/concerns/training_management.rb]: Agreed
